### PR TITLE
daily.sh: post-pull, generate vue-i18n translations and updated json language files

### PR DESCRIPTION
**Pull request by josephtingiris**
* When language files are modified, it requires manual intervention for
the changes to appear on the GUI.

* This patches daily.sh to automatically run lnms to generate the updated
json language files and vue-i18n translations.

### CODE REVIEW

**Comment by PipoCanaja** [comment]
Hi @josephtingiris 
We need to check that a LibreNMS instance out of the box has all the prerequisites to run this. If not, daily.sh will fail for a lot of instances.

**Comment by josephtingiris** [comment]
I understand.  Works for me!  haha.  I thought this one might take a while & is why I did a separate PR for mix-manifest.json.  https://github.com/librenms/librenms/pull/11019 

daily.sh complains about mix-manifest.json, every day, because the current version in the repo doesn't match the repo's language files.  Both changes would keep the manifest in sync with the language files.  Only this one would be needed if mix-manifest.json was removed from the repo and ignored.

I've tested it on more than one instance, though all my instances are basically the same.

**Comment by PipoCanaja** [comment]
if you are in sync with librenms master, then the JSON and language files are matching. Probably you changed a translation in your server and resulted in a mismatch. 

That being said, I am not familiar enough with this part of the code and the prerequisites to merge this change, so the PR will stay here a while waiting for review I'm afraid.

**Comment by josephtingiris** [comment]
> if you are in sync with librenms master, then the JSON and language files are matching.

Currently, `mix-manifest.json` is not in sync with the language files in librenms master.  Please double check.  I have.  See output below.

When/if language files are changed in master (which they have been recently) then something needs to run translation:generate (and vue-i18n:generate) to keep the individual instances' local files in sync.  If that doesn't happen then existing installs do not see the language updates on their pages.   Automating that is the purpose of this PR.  

Following is the output of a fresh clone.  When `lnms translation:generate` is run, it ultimately modifies `mix-manifest.json` and updates (installed) JSON language files for use on the from end.

```sh
[jtingiris@lt ~/src]$ date
Mon 20 Jan 2020 08:01:03 AM EST
[jtingiris@lt ~/src]$ git clone https://github.com/librenms/librenms
Cloning into 'librenms'...
remote: Enumerating objects: 18, done.
remote: Counting objects: 100% (18/18), done.
remote: Compressing objects: 100% (18/18), done.
remote: Total 151639 (delta 4), reused 0 (delta 0), pack-reused 151621
Receiving objects: 100% (151639/151639), 160.32 MiB | 28.90 MiB/s, done.
Resolving deltas: 100% (104696/104696), done.
Updating files: 100% (10715/10715), done.

[jtingiris@lt ~/src]$ cd librenms

[jtingiris@lt ~/src/librenms]$ git status
On branch master
Your branch is up to date with 'origin/master'.

nothing to commit, working tree clean

[jtingiris@lt ~/src/librenms]$ ./scripts/composer_wrapper.php install --no-dev
# output cut for brevity

[jtingiris@lt ~/src/librenms]$ git status
On branch master
Your branch is up to date with 'origin/master'.

nothing to commit, working tree clean
[jtingiris@lt ~/src/librenms]$ lnms translation:generate
[jtingiris@lt ~/src/librenms]$ git status
On branch master
Your branch is up to date with 'origin/master'.

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
        modified:   html/mix-manifest.json

no changes added to commit (use "git add" and/or "git commit -a")

[jtingiris@lt ~/src/librenms]$ git diff
diff --git a/html/mix-manifest.json b/html/mix-manifest.json
index 849a2ab1b..438dd7587 100644
--- a/html/mix-manifest.json
+++ b/html/mix-manifest.json
@@ -3,10 +3,10 @@
     "/css/app.css": "/css/app.css?id=17e56994706c74ee9663",
     "/js/manifest.js": "/js/manifest.js?id=3c768977c2574a34506e",
     "/js/vendor.js": "/js/vendor.js?id=8903cec9b99453318869",
-    "/js/lang/de.js": "/js/lang/de.js?id=04de715032d1fe1584d9",
-    "/js/lang/en.js": "/js/lang/en.js?id=e7ab0d25d61656809fe8",
-    "/js/lang/fr.js": "/js/lang/fr.js?id=51f0ee3b59a7dace8913",
+    "/js/lang/de.js": "/js/lang/de.js?id=048242b33756e80172c8",
+    "/js/lang/en.js": "/js/lang/en.js?id=4810ba45dac688ba9945",
+    "/js/lang/fr.js": "/js/lang/fr.js?id=e7e107f47cac2f5f17e0",
     "/js/lang/ru.js": "/js/lang/ru.js?id=d1a4a7e38c1e19a9f35f",
     "/js/lang/uk.js": "/js/lang/uk.js?id=a4f38c7e0cfec6593e8e",
-    "/js/lang/zh-TW.js": "/js/lang/zh-TW.js?id=d973da2eac4a300af36d"
-}
+    "/js/lang/zh-TW.js": "/js/lang/zh-TW.js?id=a5e26f8189f348d58c71"
+}
\ No newline at end of file
```
The `mix-manifest.json` in master is out of date and does not match the language files that are in master.

**Comment by SourceDoctor** [comment]
Not sure if it's good to let generating language files to the instances. I see many potential problems coming with.
Also as i remember, to run `lnms translation:generate` successfully `npm` needs to be installed, which in many cases is not such handy to be installed and uses many dependencies which are coming with it.
The actual way is the most comfortable one for the instances, i think. Successfully built and running languages files are delievered and no problems with them.

**Comment by josephtingiris** [comment]
npm nor anything special needs to be installed.  They'll generate on a vanilla LibreNMS install, without --no-dev etc.

They're generated by `vendor/martinlindhe/laravel-vue-i18n-generator` which gets installed via composer.

@Jellyfrog and I went through this in more detail on discord.  There's a bug in laravel-vue-18n that he submitted a patch for, upstream.  The bug affects the order or the arrays within.  They're non-deterministic and sort differently on different machines. That's what leads to mix-manifest being different on every install.  ? left was whether to include mix-manifest.json in the repo at all.

That aside, the bigger issue is `/opt/librenms/html/js/lang/.gitignore` and the language javascripts within that directory *never* get regenerated _unless_ `lnms translation:generate` is run.  The UI uses those files.

So, when a language resource in `/opt/librenms/resources/lang` is modified, `lnms translation:generate' *must* be run to affect those changes on the UI. 

Bottom line is when `/opt/librenms/resources/lang` is modified and checked in, lots of installs in the field will _not_ get the updates unless `lnms translation:generate` is manually run.

That's why I proposed running it in daily.sh

**Comment by josephtingiris** [comment]
FWIW, this is not  breaking change.  I hope others will test.

**Comment by josephtingiris** [comment]
Here's a different example; https://github.com/librenms/librenms/pull/10928

If `lnms translation:generate` was automated, or if `/opt/librenms/html/js/lang/.gitignore` were removed and the javascripts committed, then it would work.

**Comment by PipoCanaja** [comment]
@josephtingiris All right, let's try to get some feedback on this PR. If tests do not show any issue, then we'll merge this. 
Making easier and more predictable those translations would be good !

**Comment by murrant** [comment]
Right now we cannot do this.  The changes that update mix-manifest.json should include updated hashes when they are merged so the UI should pull in the new files.

We either need to generate mix-manifest.json in daily (including for all js files, not just the language files) and remove it from git or leave it as is now.

Currently, if we merged this it would update mix-manifest.json in daily and probably block updates... not a good thing.
